test/ILLink.RoslynAnalyzer.Tests/TestChecker.cs

```diff
@@ -137,6 +151,11 @@ private bool TryValidateExpectedWarningAttribute (AttributeSyntax attribute, Lis
 			if (!expectedWarningCode.StartsWith ("IL"))
 				throw new InvalidOperationException ($"Expected warning code should start with \"IL\" prefix.");
 
+			if (!SupportedDiagnosticIds.Contains (expectedWarningCode)) {
+				unsupportedDiagnostic = true;
+				return false;
```

Comment:
I think this approach makes it too easy to miss potential coverage holes (since it'll let tests pass if we forget to implement a particular warning ID or if there are interactions between the analyzers). I wonder if eventually we'll want to run these tests with all trim-related Roslyn analyzers. For now, could we add a new enum value to ProducedBy and use that instead for the warnings not produced by the COM analyzer?

Reply:
Good point. I do think that at some point we'll want to run tests with multiple/all Roslyn analyzers.
> For now, could we add a new enum value to ProducedBy and use that instead for the warnings not produced by the COM analyzer?

Sure thing. I think it would make sense to have each of the analyzers in that enum.